### PR TITLE
Validate if tests in test/e2e/common/storage/volume.go are still needed

### LINKED ISSUE #99468: Validate if tests in test/e2e/common/storage/volume.go re still needed
Context
@msau42 said that it's possible they now duplicate logic in other tests

/sig storage
/kind cleanup

### PR DESCRIPTION
**Pull request by Kappie37**
#### What type of PR is this?
/kind cleanup
/sig storage

#### What this PR does / why we need it:

Checks if the tests in test/e2e/common/storage/volume.go are still needed

#### Which issue(s) this PR fixes:

Fixes #99468

#### Special notes for your reviewer:

The answer is yes, they are still needed. I looked at all the other tests and found no tests that use any of the logic.
While searching for the test logic in other tests I also found the following:

Remove comment:
```
// GlusterFS test is duplicated from test/e2e/volumes.go.  Any changes made there
// should be duplicated here
```
There is no file test/e2e/volumes.go
I could also not the mentioned tests anywhere. This must mean that this file is the only place where this logic is tested.

On line 97
```
// Must match content of test/images/volumes-tester/nfs/index.html
```
directory `test/images/volumes-tester` does not exists, `test/images/volume` does.

The same goes for line 146
```
// 'test_vol' comes from test/images/volumes-tester/gluster/run_gluster.sh
```
Additionally, I could not find `test_vol` in that file (`test/images/volume/gluster/run_gluster.sh`)

The same goes for line 151 
```
// Must match content of test/images/volumes-tester/gluster/index.html
```
(content still matched)

The same goes for the comment
```
// Must match content of test/images/volumes-tester/gluster/index.html
```
(content still matched)

In file `test/e2e/storage/testsuite/volumes.go` the following comment:
```
// test/e2e/common/volumes.go duplicates the GlusterFS test from this file.  Any changes made to this
// test should be made there as well.
```
is therefore also not needed anymore

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/common/storage/volumes.go [hunk 1]
```diff
@@ -56,7 +53,6 @@ import (
 	"github.com/onsi/ginkgo"
 )
 
-// TODO(#99468): Check if these tests are still needed.
```
Comment (msau42) [inline test/e2e/common/storage/volumes.go:59]: We have "should store data" tests that run against nfs and gluster:
https://k8s-testgrid.appspot.com/sig-storage-kubernetes#gce&include-filter-by-regex=should%20store%20data&include-filter-by-regex=nfs%7Cgluster

https://github.com/kubernetes/kubernetes/blob/2170bba47a123ce823284d724cf673349c6aa378/test/e2e/storage/testsuites/volumes.go#L156

I think they are pretty similar and we could potentially get rid of the ones under "common/storage/volumes.go". I think the only thing that may be missing is a config for nfs v3. We may have to add another config under https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/drivers/in_tree.go#L95

Comment (Kappie37) [reply test/e2e/common/storage/volumes.go:59]: Thank you for your reaction!

To verify, you suggest 

* removing the tests from `test/e2e/common/storage/volumes.go`
* And one of two options:
1. Adding a test function with config for nfs v3 at `test/e2e/storage/drivers/in_tree.go` (at the bottom of the file?).
2. Adding a test config for v3 to `SupportedFsType` at `func InitNFSDriver() storageframework.TestDriver` at line 95 from `test/e2e/storage/drivers/in_tree.go` .

Is this correct? If so, I'll directly do this and update the PR.

**Comment by Kappie37** [comment]
I made an error with pushing a commit not related to this PR. Is there a way for me to revert this (and the labels etc)?

File: test/e2e/common/storage/volumes.go (hunk 1 above)
Comment (msau42) [reply test/e2e/common/storage/volumes.go:59]: Yeah, I would go with the first approach for adding a nfsv3 driver config

**Pull request closed without merging**